### ISSUE TITLE
DeclarePublicAPIAnalyzer Exception RS0022

### ISSUE THREAD

**Issue by AdamSpeight2008** [dotnet/roslyn#3626]
```
Severity	Code	Description	Project	File	Line
Message	AD0001	The User Diagnostic Analyzer 'Roslyn.Diagnostics.Analyzers.ApiDesign.DeclarePublicAPIAnalyzer' threw an exception with message 'Reported diagnostic with ID 'RS0022' is not supported by the analyzer.
Parameter name: diagnostic'.	CodeAnalysis		1
Message	AD0001	The User Diagnostic Analyzer 'Roslyn.Diagnostics.Analyzers.ApiDesign.DeclarePublicAPIAnalyzer' threw an exception with message 'Reported diagnostic with ID 'RS0022' is not supported by the analyzer.
Parameter name: diagnostic'.	BasicCodeAnalysis		1
```

**Comment by tmeschter** [comment]
@AdamSpeight2008 Yes, I see the problem. The DeclarePublicAPIAnalyzer tried to report error RS0022, but failed to tell the compiler ahead of time that it supported that error.

**Comment by gafter** [comment]
Fixed in #3635.
